### PR TITLE
test: deactivate venv after tox runs

### PR DESCRIPTION
**Pull request by noahdesu**
Signed-off-by: Noah Watkins <nwatkins@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by noahdesu** [comment]
waiting to see if this fix works for the problem in #24445

File: src/pybind/mgr/insights/run-tox.sh [hunk 1]
```diff
@@ -18,6 +19,7 @@ unset PYTHONPATH
 export CEPH_BUILD_DIR=$CEPH_BUILD_DIR
 
 source ${MGR_INSIGHTS_VIRTUALENV}/bin/activate
+trap "deactivate" EXIT
```
Comment (tchaikov) [inline src/pybind/mgr/insights/run-tox.sh]: but, i don't think the bash environmental settings configured by `${MGR_INSIGHTS_VIRTUALENV}/bin/activate` will survive the termination of the bash process to launch this `run-tox.sh` script test.

so this `trap` will not help to cleanup the test environment.

Comment (noahdesu) [reply src/pybind/mgr/insights/run-tox.sh]: @tchaikov locally testing, what you say is true. but in #24445 it looks very clearly that the test environment is leaking. i'm wondering if the test infrastructure is invoking things in a way that causes that leak.

Comment (tchaikov) [reply src/pybind/mgr/insights/run-tox.sh]: what leaked is but the log message left by the insights tox test -- /home/jenkins-build/build/workspace/ceph-pull-requests/build/py27/log/py27-0.log .

as all py27 env of tox tests share the same log file.

Comment (tchaikov) [reply src/pybind/mgr/insights/run-tox.sh]: i think that the py27 env is reused. as both ansible and insights are using the same tox `toxworkdir` and the same envlist. and in tox, envs are supposed to be reused.

Comment (noahdesu) [reply src/pybind/mgr/insights/run-tox.sh]: thanks a lot for looking at this. it still isn't clear why insights/ansible tests have an issue, but when insights was added dashboard/insights didn't have the same issue.

**Pull request closed without merging**